include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -165,39 +170,22 @@ inline void StatsdClient::send(const std::string& key,
         return;
     }
 
-    // Prepare the buffer and include the sampling rate if it's not 1.f
-    const bool needs_dot = !m_prefix.empty() && !key.empty();
-    std::string buffer(256, '\0');
-    int string_len = -1;
-    if (isFrequencyOne) {
-        // Sampling rate is 1.0f, no need to specify it
-        string_len = std::snprintf(&buffer.front(),
-                                   buffer.size(),
-                                   "%s%s%s:%d|%s",
-                                   m_prefix.c_str(),
-                                   needs_dot ? "." : "",
-                                   key.c_str(),
-                                   value,
-                                   type.c_str());
-    } else {
-        // Sampling rate is different from 1.0f, hence specify it
-        string_len = std::snprintf(&buffer.front(),
-                                   buffer.size(),
-                                   "%s%s%s:%d|%s|@%.2f",
-                                   m_prefix.c_str(),
-                                   needs_dot ? "." : "",
-                                   key.c_str(),
-                                   value,
-                                   type.c_str(),
-                                   frequency);
+    // Format the stat message
+    m_buffer.clear();
+    m_buffer.append(m_prefix);
+    if (!m_prefix.empty() && !key.empty()) m_buffer.push_back('.');
+    m_buffer.append(key);
+    m_buffer.push_back(':');
+    m_buffer.append(std::to_string(value));
+    m_buffer.push_back('|');
+    m_buffer.append(type);
+    if (frequency < 1.f) {
+        m_buffer.append("|@0.");
+        m_buffer.append(std::to_string(static_cast<int>(frequency * 100)));
```

Comment:
Cosmetic but makes it easier to read imo
```suggestion
    m_buffer.clear();
    
    m_buffer.append(m_prefix);
    if (!m_prefix.empty() && !key.empty()) {
        m_buffer.push_back('.');
    }
    
    m_buffer.append(key);
    m_buffer.push_back(':');
    m_buffer.append(std::to_string(value));
    m_buffer.push_back('|');
    m_buffer.append(type);
    
    if (frequency < 1.f) {
        m_buffer.append("|@0.");
        m_buffer.append(std::to_string(static_cast<int>(frequency * 100)));
```